src/platform/forms/save-in-progress/helpers.js

```diff
@@ -8,6 +9,33 @@ import RoutedSavableReviewPage from './RoutedSavableReviewPage';
 import FormSaved from './FormSaved';
 import SaveInProgressErrorPage from './SaveInProgressErrorPage';
 
+/**
+ * Change the heading level that is rendered inside the saveInProgressIntro.
+ * This is used inside the SaveInProgressIntro component
+ * @param headingLevel {string} - The string of what heading level you want ('h1', 'h2', etc). If you want default h3 pass in null
+ * @param appType {string} - The App Type for your application. This is set in the formConfig file for your form
+ *
+ */
+export function changeSaveInProgressHeadingLevel(headingLevel, appType) {
+  const content = (
+    <>
+      Save time—and save your work in progress—by signing in before starting
+      your {appType}
+    </>
+  );
+  const headingClass = 'usa-alert-heading';
+  switch (headingLevel) {
+    case 'h1':
+      return <h1 className={headingClass}>{content}</h1>;
+    case 'h2':
+      return <h2 className={headingClass}>{content}</h2>;
+    case 'h3':
+      return <h3 className={headingClass}>{content}</h3>;
+    default:
+      return <h3 className={headingClass}>{content}</h3>;
+  }
+}
+
```

Comment:
I think this is a bit overkill - there are a few simpler ways of accomplishing the same goal without this switch statement.

## Functional component

Here is an example of a [functional component being used in a clever way to have dynamic heading levels](https://github.com/department-of-veterans-affairs/component-library/blob/00571e0bf49c697b99569096b35479561c6ccd7a/src/components/AlertBox/AlertBox.jsx#L99-L109):

```jsx
const H = `h${this.props.level}`;

return <H>My heading</H>
```

## React.createElement

React also provides a [`createElement` function](https://reactjs.org/docs/react-api.html#createelement) which could be used here. You can see [an example of it in use in `vets-website`](https://github.com/department-of-veterans-affairs/vets-website/blob/809a8c25d28c335538cd30b0f85f0945042eed1c/src/platform/forms/components/SignInLink.jsx#L41-L48).

Reply:
@bkjohnson yeah, initially I was thinking having the logic spelled out made it more clear what was happening here but I think I agree using a functional component might be a better way to go. Let me refactor.